### PR TITLE
Use HttpMethod instead of String

### PR DESCRIPTION
**Pull request by daschult**

### CODE REVIEW

File: client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java [hunk 1]
```diff
@@ -471,11 +472,11 @@ private boolean isFlowableByteArray(TypeToken entityTypeToken) {
     private Maybe<?> handleBodyReturnTypeAsync(final HttpResponse response, final SwaggerMethodParser methodParser, final Type entityType) {
         final TypeToken entityTypeToken = TypeToken.of(entityType);
         final int responseStatusCode = response.statusCode();
-        final String httpMethod = methodParser.httpMethod();
+        final HttpMethod httpMethod = methodParser.httpMethod();
         final Type returnValueWireType = methodParser.returnValueWireType();
 
         final Maybe<?> asyncResult;
-        if (httpMethod.equalsIgnoreCase("HEAD")
+        if (httpMethod == HttpMethod.HEAD
```
Comment (RikkiGibson) [inline client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java]: So does ExpandableStringEnum use an interned instance when `.fromString()` is called?

File: client-runtime/src/test/java/com/microsoft/rest/v2/http/HttpMethodTests.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    @Test
+    public void HEAD()
+    {
+        assertEquals("HEAD", HttpMethod.HEAD.toString());
+    }
+
+    @Test
+    public void fromStringWithExisting()
+    {
+        assertSame(HttpMethod.POST, HttpMethod.fromString("POST"));
```
Comment (RikkiGibson) [inline client-runtime/src/test/java/com/microsoft/rest/v2/http/HttpMethodTests.java]: 👍

File: client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java (hunk 1 above)
Comment (daschult) [reply client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java]: Yeah. It checks to see if an instance already exists for the provided String value and will return it if so.

**Comment by daschult** [comment]
CI failure is just a timeout issue. Everything passes locally.

**Pull request merged**